src/app/main/services/recipes-service/recipes.service.ts

```diff
@@ -0,0 +1,58 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable, Subscribable, Subscription, map, mergeAll, of, tap } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Recipe } from 'src/interfaces/recipe';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecipesService {
+  emptyRecipe: Recipe = {
+    id: '',
+    author: '',
+    categories: ['dinner'],
+    cookingTime: {hours: 0, minutes: 0},
+    date: '',
+    description: '',
+    ingredients: [{name: '', quantity: ''}],
+    isVegan: false,
+    picture: '',
+    portionsNumber: 0,
+    rating: 0,
+    steps: [''],
+    title: ''
+  }
+  private _recipes$ : BehaviorSubject<Recipe[]> = new BehaviorSubject([this.emptyRecipe]);
+  private recipes$ = this._recipes$.asObservable();
```

Comment:
private  readonly _r....
(please fix it on other places too)

and i see that this object link go outside of this service so it can be rewritten , it's dangerous

Reply:
I can't make this variable "private _recipes$" readonly, because I need to put data inside it (with rxjs next method) once recipes are received. 

Could you pls explain where exactly this object goes outside the service? As I see, it is private, so can't be assessed from any other place.